tests/bsim/bluetooth/mesh/src/test_beacon.c

```diff
@@ -1149,18 +1154,13 @@ static void private_beacon_create(struct net_buf_simple *buf, const uint8_t *net
 		FAIL("Unable to generate Net ID");
 	}
 
-	err = bt_mesh_private_beacon_key(net_key, priv_beacon_key);
+	err = bt_mesh_private_beacon_key(net_key, &priv_beacon_key);
```

Comment:
Should `bt_mesh_key_destroy` be called after encrypting the beacon?

Reply:
It is used for decryption of incoming beacons here: https://github.com/alxelax/zephyr/blob/refactor_key_usage_2/tests/bsim/bluetooth/mesh/src/test_beacon.c#L355-L356